pkg/metrics/async.go

```diff
@@ -0,0 +1,46 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package metrics
+
+import (
+	"context"
+
+	"github.com/abcxyz/pkg/logging"
+)
+
+// asyncFunctionCall handles the async part of SendMetricRequest, but accepts
+// a function other than SendMetricRequestSync for testing.
+func asyncFunctionCall(ctx context.Context, funcToCall func() error) func() {
```

Comment:
Should the return func return an error?

```suggestion
func asyncFunctionCall(ctx context.Context, funcToCall func() error) func() error {
```

Reply:
I don't think so in this case. All it does is block until goroutine is done.

I guess I could send the error returned by funcToCall in a channel back to this callback, though I'm not sure if that is a common pattern or not.

Reply:
That, plus any context errors (L41)